state.go

```diff
@@ -317,6 +292,66 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string) (bool, time.Duration, error) {
```

Comment:
My intent was for `error != nil` to indicate an error transmitting the ping - for example, the node name not being present in the list of members.  The `success` bool indicates whether or not a response was received within `m.config.ProbeTimeout`.  I think keeping both the bool and the error string is valuable since they differentiate between a fatal error that would prevent the ping from being sent, and a potential transient error that prevents the target from responding.  Perhaps renaming `success` to `responseReceived` would clarify the meaning.

One use case for the keeping the bool and error string return values can be seen in the Consul modifications that invoke Memberlist.Ping().  (See https://github.com/jsullivan3/consul/pull/2 for details.)  This patch will continue issuing ping packets to the target node if no response is received, but will not attempt another ping if `error != nil`.